core/include/detray/navigation/intersection/intersection.hpp

```diff
@@ -38,26 +41,25 @@ struct intersection2D {
     using transform3_type = dtransform3D<algebra_t>;
     using nav_link_t = typename surface_descr_t::navigation_link;
 
-    /// Descriptor of the surface this intersection belongs to
-    surface_descr_t sf_desc{};
-
-    /// Local position of the intersection on the surface
-    point3_type local{detail::invalid_value<T>(), detail::invalid_value<T>(),
-                      detail::invalid_value<T>()};
-
-    /// Distance between track and candidate
-    scalar_type path = detail::invalid_value<T>();
-
-    /// Navigation information (next volume to go to)
-    nav_link_t volume_link{detail::invalid_value<nav_link_t>()};
-
     /// Result of the intersection (true = inside, false = outside)
     bool_t status{false};
 
     /// Direction of the intersection with respect to the track (true = along,
     /// false = opposite)
     bool_t direction{true};
```

Comment:
Perhaps these can also go at the back

Reply:
Yes, good point. I shuffled it around until the alignment fits, but sorting by size is indeed easier. I can already see that the struct has not padding anymore (24 bytes)

Reply:
Are you saying that you have these members:

```
bool_t status // A (1 byte)
bool_t direction // B (1 byte)
nav_link_t volume_link // C (4 bytes)
scalar_type path // D (4 bytes)
surface_descr_t // E (4 bytes)
 ```
 
 And they are aligned like this:
 
 ```
 AB......
 CCCCDDDD
 EEEE....
 ```

Or am I misunderstanding?

Reply:
Almost, ```nav_link_t``` has 2 bytes (```std::uint_leat16_t```) and the ```surface_descriptor``` currently has 16 bytes (but is composed of smaller types and should have their alignment, right?)

Reply:
It seems that, indeed, `surface_descriptor` is aligned to 8 bytes; which makes sense.

However, I am getting to a minimum size of 36 bytes; where did you get 24? I.e. 16 byte surface descriptor, 12 byte vector3, 4 byte scalar, 2 byte navigation link, and two 1 byte bools.

Reply:
The vector3 is only added if ```do_debug``` is true, which is after this PR only the case in the tests and validations. In the nominal propagation this is currently not needed